src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Flow.java

```diff
@@ -2216,13 +2218,22 @@ boolean isFinalUninitializedStaticField(VarSymbol sym) {
          *  index into the vars array.
          */
         void newVar(JCVariableDecl varDecl) {
-            VarSymbol sym = varDecl.sym;
+            newVar(varDecl, varDecl.sym);
+        }
+
+        /** Initialize new trackable variable by setting its address field
+         *  to the next available sequence number and entering it under that
+         *  index into the vars array.
+         */
+        void newVar(JCTree pos,VarSymbol sym) {
```

Comment:
This is used only twice. One from `void newVar(JCVariableDecl varDecl)` which is very intuitive and one with `newVar(null, component);` which I understand. But, is there any reason to create a `var` in the future with something else than `null` (unrelated to `sym`?). Maybe the comment needs to be updated to document what should be the relation (if any) between `pos` and `sym`.

Reply:
Oops, that's actually an oversight - there should be a pos specified in all cases. Should be fixed now.